crates/transaction-pool/src/pool/best.rs

```diff
@@ -196,6 +198,30 @@ impl<T: TransactionOrdering> Iterator for BestTransactions<T> {
     }
 }
 
+impl<T: PoolTransaction<Consensus: Into<TransactionSignedEcRecovered>>> PayloadTransactions
+    for Box<dyn crate::BestTransactions<Item = Arc<ValidPoolTransaction<T>>>>
+{
+    fn next(&mut self, _ctx: ()) -> Option<TransactionSignedEcRecovered> {
+        Iterator::next(self).map(|tx| tx.to_recovered_transaction())
+    }
+
+    fn mark_invalid(&mut self, _sender: Address, _nonce: u64) {
+        // TODO: Implement this. Can't use BestTransactions::mark_invalid directly
+        //   because it requires a sender ID.
+    }
+}
```

Comment:
I think I'd prefer not to mix these traits for now, and instead introduce a simple helper type that tracks invalid senders in a HashSet instead, this would unblock the pr entirely.

like

```rust
struct BestPayloadTransaction<I> {
  invalid: HashSet<Address>,
  best: I
}
```

Reply:
agree, updated, please check if that's what you had in mind.